agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileCache.java

```diff
@@ -61,4 +67,37 @@ String poll() {
   Queue<String> getPersistedFilesCache() {
     return persistedFilesCache;
   }
+
+  @Nullable
+  private static List<File> sortPersistedFiles(File folder) {
+    Collection<File> files = FileUtils.listFiles(folder, new String[] {"trn"}, false);
+    Comparator<File> comparator =
+        new Comparator<File>() {
+          @Override
+          public int compare(File file1, File file2) {
+            long file1MilliSeconds =
+                Long.parseLong(file1.getName().substring(0, file1.getName().lastIndexOf('-')));
+            long file2MilliSeconds =
+                Long.parseLong(file2.getName().substring(0, file2.getName().lastIndexOf('-')));
+            if (file1MilliSeconds > file2MilliSeconds) {
+              return 1;
+            } else if (file1MilliSeconds < file2MilliSeconds) {
+              return -1;
+            }
+
+            return 0;
+          }
+        };
+
+    List<File> result = new ArrayList<File>(files);
+    try {
+      Collections.sort(result, comparator);
+    } catch (RuntimeException ex) {
+      LoggerFactory.getLogger(LocalFileCache.class)
+          .error("Fail to sort a list of persisted files on disk", ex);
+      return null;
+    }
+
+    return result;
+  }
```

Comment:
I think ok to keep it simple and just sort by `lastModified`, may not be perfect, but good enough since nothing critically bad will happen if timestamps are incorrect

```suggestion
  private static List<File> sortPersistedFiles(File folder) {
    return FileUtils.listFiles(folder, new String[] {"trn"}, false).stream()
        .sorted(Comparator.comparing(File::lastModified))
        .collect(Collectors.toList());
  }
```

or, for the more complex implementation, something like this:

```suggestion
  private static List<File> sortPersistedFiles(File folder) {
    return FileUtils.listFiles(folder, new String[] {"trn"}, false).stream()
        .sorted(Comparator.comparing(LocalFileCache::timestampFromFileName))
        .collect(Collectors.toList());
  }

  // visible for testing
  static long timestampFromFileName(File file) {
    int endIndex = file.getName().lastIndexOf('-');
    if (endIndex == -1) {
      return 0;
    }
    String timestamp = file.getName().substring(0, endIndex);
    try {
      return Long.parseLong(timestamp);
    } catch (NumberFormatException e) {
      return 0;
    }
  }
```